apps/web/app/(app)/environments/[environmentId]/settings/api-keys/actions.ts

```diff
@@ -1,10 +1,20 @@
 "use server";
 
+import { authOptions } from "@/app/api/auth/[...nextauth]/authOptions";
+import { canUserAccessApiKey } from "@/lib/authorization";
 import { deleteApiKey, createApiKey } from "@formbricks/lib/services/apiKey";
 import { TApiKeyCreateInput } from "@formbricks/types/v1/apiKeys";
+import { getServerSession } from "next-auth";
 
 export async function deleteApiKeyAction(id: string) {
-  return await deleteApiKey(id);
+  const session = await getServerSession(authOptions);
+  const authorized = await canUserAccessApiKey(session, id);
```

Comment:
is it maybe possible to have a generic method `canUserAccessRessource` that works not only for ApiKeys but also for e.g. surveys? Any ideas? 😊
If not possible please move the authorization method to the api key service.

Reply:
tried the generic method but it wasn't looking promising and each resource would have a very different take! hence deciding to keep them all separate!